ts/src/management/tunnelManagementHttpClient.ts

```diff
@@ -30,6 +30,32 @@ function comparePorts(a: TunnelPort, b: TunnelPort) {
     return (a.portNumber ?? Number.MAX_SAFE_INTEGER) - (b.portNumber ?? Number.MAX_SAFE_INTEGER);
 }
 
+function parseDate(value?: string | Date) {
+    return typeof value === 'string' ? new Date(Date.parse(value)) : value;
+}
+
+/**
+ * Fixes Tunnel properties of type Date that were deserialized as strings.
+ */
+function parseTunnelDates(tunnel: Tunnel | null) {
+    if (!tunnel) return;
+    tunnel.created = parseDate(tunnel.created);
+    if (tunnel.status) {
+        tunnel.status.lastHostConnectionTime = parseDate(tunnel.status.lastHostConnectionTime);
+        tunnel.status.lastClientConnectionTime = parseDate(tunnel.status.lastClientConnectionTime);
+    }
+}
+
+/**
+ * Fixes TunnelPort properties of type Date that were deserialized as strings.
+ */
+function parseTunnelPortDates(port: TunnelPort | null) {
```

Comment:
`port?:`

Reply:
It should be an object or null, never undefined, because this is a result object that is being returned to the caller of the API.